From 7a8c50f620c7484af9d750f484df8a6837e6b2a5 Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Wed, 17 Oct 2012 15:27:03 -0400
Subject: [PATCH 19/42] [daemon] Make sure inpe is initialized before all
 error handling.

find_certificate() and set_up_inpe() errors wind up being at the same
place, which means when find_certificate is called, inpe already must be
NULL.

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 src/daemon.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/daemon.c b/src/daemon.c
index 69aea4b..534fb23 100644
--- a/src/daemon.c
+++ b/src/daemon.c
@@ -362,6 +362,7 @@ handle_signing(context *ctx, struct pollfd *pollfd, socklen_t size,
 	struct iovec iov;
 	ssize_t n;
 	char *buffer = malloc(size);
+	Pe *inpe = NULL;
 
 	if (!buffer) {
 oom:
@@ -433,7 +434,6 @@ malformed:
 		goto finish;
 	}
 
-	Pe *inpe = NULL;
 	rc = set_up_inpe(ctx, infd, &inpe);
 	if (rc < 0)
 		goto finish;
-- 
1.7.12.1

